ui/desktop/app/routes/scopes/scope/projects/sessions/session.js

```diff
@@ -12,6 +12,11 @@ export default class ScopesScopeProjectsSessionsSessionRoute extends Route {
   @service store;
 
   // =methods
+  resetController(controller, isExiting, transition) {
+    if (isExiting && transition.targetName !== 'error') {
```

Comment:
why do we need to reset the controller explicitly here? Wouldn't this automatically happen when we change routes? 🤔

Reply:
As Carlos mentioned above, I am going to add a comment for this but I am going to give more details here. Controllers are [singletons](https://guides.emberjs.com/v4.4.0/routing/controllers/#toc_where-and-when-to-use-controllers). So `isRawApiVisible` will be shared among all instances of that route. For example, when I toggle the view to see the API JSON for the credential, then transition away from that screen, the `isRawApiVisible` property is still `true`. So if I were to go back to another session details screen, the API JSON view would automatically be visible even though it shouldn't. So Ember has the [resetController](https://api.emberjs.com/ember/4.4/classes/Route/methods/resetController?anchor=resetController) hook that allows you to reset the value to whatever you want when the user transitions away from that route.

Reply:
Can you explain why this needs to be in the controller in the first place? It's not obvious to me why this is the case

Reply:
I decided to put this in the controller because this is not part of the model data and only used to toggle how the user views the credential UI. Is there a better place this should live?

Reply:
Is there a reason this couldn't just live in the backing class of the template?

Reply:
Going to move the credential panel to a component after chatting offline.

Reply:
Done.